holidayextras/brands
# Globals misc updates

#### What does this PR do? (please provide any background)
* Added `.pos-abs` and `.pos-rel` utility classes for handy positioning.
* Added `.fill` class for making elements fill relative space.
* Added `@navbar-margin-bottom` to affix-spacer.
* Added appropriate z-index to affixed navbars.
* Added `.text-default` helper class for resetting color to default `<body>` text color.
* Added color fix for `.label-default` to ensure readability on lighter label backgrounds.
* Various additions to improve usability of targets of Bootstrap's collapse plugin.
  * Added `.collapse-icon` class for disclosure icons in collapse target.
    * Also added custom '+' icon for use as collapse icons - e.g. `<span class="i-plus">+</span>`.
    * Currently supporting `.i-plus`, `.fa-angle-down` and `.fa-angle-up` icons.
  * Added `.collapse-heading` class for low-lighting collapse targets when collapsed.
  * Added `.collapsed-text` and `.expanded-text` for toggling appropriate text in collapse targets.
* Added a `.text-faded` class and corresponding `@text-faded` variable for even moar muted txt.
  * Also tweaked `@text-muted` default to calculate value from `@text-color`, with a more conservative level of transparency.

#### How can this be tested?
`cd pattern-library && npm link ../brands/globals`
`git pull pl-layout-bg-variants-and-collapse-pages` - see [this corresponding PR](https://github.com/holidayextras/pattern-library/pull/252).

#### Any tech debt?
We should probably create a separate file in globals for custom icons prefixed with `.i-*`. These should ideally not rely upon a resource like an icon font file - i.e. icons that are absolutely necessary for UI functionality, like an 'x' close icon.

#### Screenshots / Screencast

#### What gif best describes how you feel about this work?
![]()


---

**Reviewer**

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Ensured the release has the correct version bump.
- Checked all the work locally on pattern library.
